### ISSUE TITLE
Conditional types don't narrow the type for Mapped types

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#21735]
cc @ahejlsberg 

**TypeScript Version:**  master

**Search Terms:** 

**Code**

```ts
type A<T> = T extends string ? { [P in T]: void; } : T;
type B<T> = string extends T ? { [P in T]: void; } : T;
```

**Expected behavior:**

pass

**Actual behavior:**

```
$ node built/local/tsc.js --noEmit index.ts
index.ts(1,40): error TS2322: Type 'T' is not assignable to type 'string'.
index.ts(2,40): error TS2322: Type 'T' is not assignable to type 'string'.
```

**Playground Link:**  

**Related Issues:**

**Comment by falsandtru** [comment]
cc @ahejlsberg

**Comment by ahejlsberg** [comment]
Now fixed in #21744. Note that `B<T>` still produces an error as expected.

**Comment by falsandtru** [comment]
Thanks for your quick work!
